### DISCUSSION TITLE
Is there a way to control capitalization for interpolated messages?

### DISCUSSION THREAD

**Discussion by janhassel** [intlify/vue-i18n#537] [Q&A]
Hey, I'm wondering if there is a way to control the capitalization of interpolated messages.

Assuming the following setup:
```js
const messages = {
  en: {
    device: 'Device',
    add: 'Add {0}',
  },
  de: {
    device: 'Gerät',
    add: '{0} hinzufügen',
  },
};
```

I'd intend to use it this way:
```js
t('add', [t('device')]);
```

which would result in the following interpolated message:

```
en: "Add Device"
de: "Gerät hinzufügen"
```

While the `de` version is correct, "Device" in the `en` version should be lowercase ("Add device" instead of "Add Device"). Is there a way to control this in the message syntax?

I'm basically looking for something like the following (based on how it works with linked messages):
```js
{
  add: 'Add @.lower:{0}',
}
```

**Comment by kazupon** [comment]
Unfortunately, the syntax of the message format syntax only supports modifiers for linked messages.
You need to make it refer to other messages as follows.

```json
{
    "device": "Device",
    "add": "Add @.lower:{'device'}"
}
```

**Reply by janhassel** [reply]
Ok, so its currently not possible to have this generic format when intending to pair with a few messages but rather I'd need to create messages like this:

```js
{
  add: 'Add',

  device: 'Device',
  connection: 'Connection',
  state: 'State',

  addDevice: '@:add @.lower:device',
  addConnection: '@:add @.lower:connection',
  addState: '@:add @.lower:state',
}
```

**Comment by janhassel** [comment]
If anyone else runs into this issue: I ended up just using a message function:

```ts
{
  add: ({ list }: MessageContext<VueMessageType>) => `Add ${String(list(0)).toLowerCase()}`,
}
```

**Comment by jonkad** [comment]
Hi, 
Is it possible to use it like this: `:label="$t('@.capitalize:emailAddress')"` or `<p>{{ $t('@.capitalize:emailAddress') }}</p>` ?

**Comment by jonkad** [comment]
Hi,

Found a Solution:

en.json:
```
{
 "cap": "@.capitalize:{t}",
  "emailAddress": "email address"
}
```
`:label="$t('cap', {t: 'emailAddress'})"`

**Comment by sneakylenny** [comment]
I got it working in version `9.3.0-beta.27`:

```jsonc
{
    // CRUD
    "Create": "Create",
    "Update": "Update",
    "Delete": "Delete",

    // Entity crud
    "Create entity": "@:Create @.lower:{entity}",
    "Update entity": "@:Update @.lower:{entity}",
    "Delete entity": "@:Delete @.lower:{entity}",
    
    // Entities
    "User": "User | Users",
}
```

And use it like this:

```vue
<template>
  <div v-text="$t('Create entity', { entity: 'User' })" />
  <div v-text="$t('Create entity', { entity: 'User' }, 2)" />
</template>
```

Which results in:

```
Create user
Create users
```

My code:

```ts
const i18n = createI18n<[MessageSchema], 'en-US'>({
  locale: userPreferences.value.lang ?? defaultLocale,
  fallbackLocale: 'en-US',
  messages: locales,

  legacy: false, // I use Vue 3

  // silentFallbackWarn: true,
  silentTranslationWarn: true
})
```

Hope this helps anyone!
